### PR TITLE
Add option to override L5 signal health and also support to enable it

### PR DESCRIPTION
**Pull request by bugobliterator**
* Since GPS L5 signal is currently marked unhealthy ublox by default doesn't use it. To allow using L5, we need to allow override the health using the L1 equivalent. This PR refers this document for this https://content.u-blox.com/sites/default/files/documents/GPS-L5-configuration_AppNote_UBX-21038688.pdf 
* Also adds support to enable L5 signal in general.

### CODE REVIEW

File: libraries/AP_GPS/AP_GPS_UBLOX.h [hunk 1]
```diff
@@ -513,7 +516,7 @@ class AP_GPS_UBLOX : public AP_GPS_Backend
     struct PACKED ubx_mon_ver {
         char swVersion[30];
         char hwVersion[10];
-        char extension; // extensions are not enabled
+        char extension[50]; // extensions are not enabled
```
Comment (tridge) [inline libraries/AP_GPS/AP_GPS_UBLOX.h:519]: why the change to extension?

**Review by tridge — CHANGES_REQUESTED** [review]

**Comment by tridge** [comment]
also should be 1 commit if you don't mind

File: libraries/AP_GPS/AP_GPS.cpp [hunk 2]
```diff
@@ -347,7 +347,7 @@ const AP_Param::GroupInfo AP_GPS::var_info[] = {
     // @Param: _DRV_OPTIONS
     // @DisplayName: driver options
     // @Description: Additional backend specific options
-    // @Bitmask: 0:Use UART2 for moving baseline on ublox,1:Use base station for GPS yaw on SBF,2:Use baudrate 115200,3:Use dedicated CAN port b/w GPSes for moving baseline,4:Use ellipsoid height instead of AMSL
+    // @Bitmask: 0:Use UART2 for moving baseline on ublox,1:Use base station for GPS yaw on SBF,2:Use baudrate 115200,3:Use dedicated CAN port b/w GPSes for moving baseline,4:Use ellipsoid height instead of AMSL, 5: Override GPS satellite health of L5 band from L1 health
```
Comment (rmackay9) [inline libraries/AP_GPS/AP_GPS.cpp:350]: nitpick, we don't need the little space between the number and description.  obviously no big deal either way.

**Comment by MallikarjunSE** [comment]
Hey @bugobliterator 
How is this handled if 2 different GPSs are used? Say, GPS1 as HerePro and GPS2 as Here4?

**Comment by tridge** [comment]
> How is this handled if 2 different GPSs are used? Say, GPS1 as HerePro and GPS2 as Here4?

in that case they are DroneCAN GPS modules, so the option is set on the CAN node, not on the flight controller

File: libraries/AP_GPS/AP_GPS.cpp (hunk 2 above)
Comment (tridge) [reply libraries/AP_GPS/AP_GPS.cpp]: fixed

**Comment by tridge** [comment]
@bugobliterator I've tested on a M10 L1/L5 and fixed the detection to work for M10 as well as F9

**Review by tridge — APPROVED** [review]
